File: algorithm/src/main/java/baekjoon/p12778/Main.java
package baekjoon.p12778;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while (t-- > 0) {
            int m = sc.nextInt();
            String c = sc.next();
            String[] inputs = new String[m];
            for (int i = 0; i < inputs.length; i++) {
                inputs[i] = sc.next();
            }

            for (String solve : solve(c, inputs)) {
                System.out.print(solve + " ");
            }
            System.out.println();
        }
    }

    private static String[] solve(String c, String[] inputs) {
        String[] ret = new String[inputs.length];
        for (int i = 0; i < ret.length; i++) {
            if (c.equals("C")) {
                char ch = inputs[i].charAt(0);
                ret[i] = String.valueOf(1 + ch - 'A');
            } else {
                int n = Integer.parseInt(inputs[i]);
                ret[i] = String.valueOf((char) ('A' + n - 1));
            }
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p12778/Main.kt
package baekjoon.p12778

fun main(args: Array<String>) {
    repeat(readLine()!!.toInt()) {
        val (m, c) = readLine()!!.split(" ")
        val inputs = readLine()!!.split(" ")
        println(solve(c, inputs).joinToString(" "))
    }
}

private fun solve(c: String, inputs: List<String>): List<String> {
    return when (c == "C") {
        true -> inputs.map { (1 + it[0].toInt() - 'A'.toInt()).toString() }
        false -> inputs.map { ('A' + it.toInt() - 1).toString() }
    }
}